app/scripts/lib/metaRPCClient.js

```diff
@@ -0,0 +1,75 @@
+import { EventEmitter } from 'events';
+import { EthereumRpcError } from 'eth-rpc-errors';
+import createRandomId from '../../../shared/modules/random-id';
+
+class MetaRPCClient {
+  constructor(connectionStream) {
+    this.connectionStream = connectionStream;
+    this.notificationChannel = new EventEmitter();
+    this.requests = new Map();
+    this.connectionStream.on('data', this.handleResponse.bind(this));
+    this.connectionStream.on('end', this.close.bind(this));
+    // eslint-disable-next-line no-constructor-return
+    return new Proxy(this, {
+      get: (object, property) => {
+        if (object[property]) {
+          return object[property];
+        }
+        return (...p) => {
+          const cb = p[p.length - 1];
+          const params = p.slice(0, -1);
+          const id = createRandomId();
+          object.requests.set(id, cb);
+          object.connectionStream.write({
+            method: property,
+            params,
+            id,
+          });
+        };
+      },
+    });
+  }
+
+  onNotification(cb) {
+    this.notificationChannel.addListener('notification', (data) => {
+      cb(data);
```

Comment:
not sure if this should be `cb(null, data)` to follow nodejs callback conventions since this only listens for notifications and would never get an error

Reply:
shouldn't need an error. call it a handler instead of a callback to avoid the confusion